lib/HeroBonus.h

```diff
@@ -27,9 +27,9 @@ typedef std::set<CBonusSystemNode*> TNodes;
 typedef std::set<const CBonusSystemNode*> TCNodes;
 typedef std::vector<CBonusSystemNode *> TNodesVector;
 
-class CSelector : std::function<bool(const Bonus*)>
+class CSelector : std::function<bool(const std::shared_ptr<Bonus>)>

```

Comment:
shared_ptr here is pointless. Selectors has nothing to do with object ownership.